docs/sync/v0.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+package treecrdt.sync.v0;
+
+// Canonical envelope for v0 messages. Concrete message definitions live in:
+// - docs/sync/v0/types.proto
+// - docs/sync/v0/ops.proto
+// - docs/sync/v0/filters.proto
+// - docs/sync/v0/messages.proto
+
+import "docs/sync/v0/messages.proto";
+
+// Envelope type for all protocol messages.
+//
+// This corresponds to the examples in docs/sync/v0.md where each message has:
+// - v (protocol version)
+// - doc_id (document identifier)
+// - exactly one payload variant set
+message SyncMessage {
+  uint32 v = 1;
+  string doc_id = 2;
+
+  oneof payload {
+    Hello hello = 10;
```

Comment:
as I understood to seperate `oneof` fields from others it has `+10` to field number. Should it be also common across all schemas.